### PR TITLE
Fix/integration

### PR DESCRIPTION
**Pull request by raonismaneoto**

### CODE REVIEW

File: utils/keys_utils.go [hunk 1]
```diff
@@ -136,26 +137,35 @@ func GeneratePrivateKey(bitSize int) (*rsa.PrivateKey, error) {
...
-	// pem.Block
+	return EncodePrivKeyToPem(privateKey), EncodePublicKeyToPem(publicKey)
+}
+
+func EncodePrivKeyToPem(privKey *rsa.PrivateKey) []byte {
+	privDER := x509.MarshalPKCS1PrivateKey(privKey)
+
 	privBlock := pem.Block{
-		Type:    "RSA PRIVATE KEY",
+		Type:    "RSA PUBLIC KEY",
```
Comment (wesleymonte) [inline utils/keys_utils.go]: The correct one would be `PRIVATE`?

File: worker/worker_test.go [hunk 2]
```diff
@@ -41,8 +41,17 @@ func TestParseWorkerConfiguration(t *testing.T) {
...
+
+	expectedWorker := Worker{
+		Vcpu:    workerTestInstance.Vcpu,
+		Ram:     workerTestInstance.Ram,
+		Id:      workerTestInstance.Id,
+		QueueId: workerTestInstance.QueueId,
+	}
 
-	if parsedWorker != workerTestInstance {
+	if parsedWorker != expectedWorker {
```
Comment (wesleymonte) [inline worker/worker_test.go:54]: Is this verification secure?

File: utils/keys_utils.go (hunk 1 above)
Comment (raonismaneoto) [reply utils/keys_utils.go]: yep, nice catch

File: worker/worker_test.go (hunk 2 above)
Comment (raonismaneoto) [reply worker/worker_test.go:54]: yep

**Review by wesleymonte — APPROVED** [review]

**Pull request merged**